functions/dropwhile.go

```diff
@@ -0,0 +1,19 @@
+package functions
+
+func (s SliceType) DropWhile(f func(ElementType) bool) (out SliceType) {
+	if f == nil {
+		return s
+	}
+	failed := false
+	for _, v := range s {
```

Comment:
I think the way that I shared it with you earlier on discord might actually be more readable. Can you update this? Thanks!